csm/conf/uds.py

```diff
@@ -0,0 +1,164 @@
+# CORTX-CSM: CORTX Management web and CLI interface.
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+from csm.conf.salt import SaltWrappers
+from pathlib import Path
+from pwd import getpwnam
+from shutil import copyfile, rmtree
+from tempfile import NamedTemporaryFile
+
+import json
+import os
+
+
+class UDSConfigGenerator:
+    HAPROXY_BEGIN_UDS = '\n# BEGIN UDS\n'
+    HAPROXY_END_UDS = '\n# END UDS\n'
+    HAPROXY_UDS_WARNING = """\
+# The following HAproxy config entries, as well as the ``# BEGIN UDS`` and
+# ``# END UDS`` comment lines surrounding it, were automatically generated by
+# ``csm_setup``. Please *do not edit these manually*.
+# Only a single occurrence of an UDS-related config block is supported by
+# ``csm_setup``.
+"""
+    HAPROXY_CONFIG_PATH = '/etc/haproxy/haproxy.cfg'
+    HAPROXY_TEMP_CONFIG_PREFIX = 'haproxy.cfg.'
+    UDS_HOME_DIR = '/var/lib/uds'
+    UDS_CONFIG_DIR = f'{UDS_HOME_DIR}/.uds'
+    UDS_CONFIG_PATH = f'{UDS_CONFIG_DIR}/uds-config.json'
+    UDS_USERNAME = 'uds'
+
+    @staticmethod
+    def generate_haproxy_config():
+        cluster_ip = SaltWrappers.get_salt_call('pillar.get', 'cluster:cluster_ip')
+        minions = list(SaltWrappers.get_salt('grains.get', 'id').values())
+        # Codacy was complaining about the use of `assert` below :(
+        # assert len(minions) == 2
+        minions.sort()
+        return f"""\
+frontend uds-frontend
+    mode tcp
+    bind localhost4:5000
+    bind localhost6:5000
+    bind {cluster_ip}:5000
+    acl udsbackendacl dst_port 5000
+    use_backend uds-backend if udsbackendacl
+
+backend uds-backend
+    mode tcp
+    balance static-rr
+    server uds-1 {minions[0]}:5000 check
+    server uds-2 {minions[1]}:5000 check"""
```

Comment:
Minor update for readability -- suggest to do 

```
    server uds-2 {minions[1]}:5000 check\
"""
```

instead of one line:

```
    server uds-2 {minions[1]}:5000 check"""
```

Reply:
Fixed.